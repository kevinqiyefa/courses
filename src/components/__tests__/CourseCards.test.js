import React from 'react';
import { shallow } from 'enzyme';
import CourseCards from '../CourseCards';

import toJson from 'enzyme-to-json';

describe('Testing CourseCards Component', () => {
  let wrapped;
  const coursesData = [
    {
      id: 111,
      subject: 'Math',
      length: 60,
      course_name: 'Algebra I',
      course_description:
        'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
      studentIDs: [2, 5]
    }
  ];

  beforeEach(() => {
    wrapped = shallow(<CourseCards courses={coursesData} />);
  });

  it('shows a card section', () => {
    expect(wrapped.hasClass('card')).toEqual(true);
  });

  it('matches snapshot', () => {
    const serialized = toJson(wrapped);
    expect(serialized).toMatchSnapshot();
  });
});
