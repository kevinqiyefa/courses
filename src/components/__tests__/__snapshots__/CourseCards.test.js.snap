// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CourseCards Component matches snapshot 1`] = `
<Link
  className="card border-primary mb-3 card-link-el"
  key="111"
  to="/111"
>
  <h2
    className="card-header"
  >
    Algebra I
  </h2>
  <div
    className="card-body"
  >
    <div
      className="course-card-subject"
    >
      <h5>
        Subject: 
        Math
      </h5>
      <h5>
        Length: 
        60
         mins
      </h5>
    </div>
    <p
      className="card-text text-left"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </p>
  </div>
</Link>
`;
